runtime/pkg/duckdbsql/duckdbsql.go

```diff
@@ -4,24 +4,163 @@ import (
 	"context"
 	databasesql "database/sql"
 	"database/sql/driver"
+	"encoding/json"
+	"regexp"
 	"sync"
 
 	"github.com/marcboeker/go-duckdb"
 )
 
+type AST struct {
+	sql       string
+	ast       astNode
+	rootNodes []*selectNode
+	aliases   map[string]bool
+	added     map[string]bool
+	fromNodes []*fromNode
+	columns   []*columnNode
+}
+
+type selectNode struct {
+	ast astNode
+}
+
+type columnNode struct {
+	ast astNode
+	ref *ColumnRef
+}
+
+type fromNode struct {
+	ast      astNode
+	parent   astNode
+	childKey string
+	ref      *TableRef
+}
+
+func Parse(sql string) (*AST, error) {
+	sqlAst, err := queryString("select json_serialize_sql(?::VARCHAR)", sql)
+	if err != nil {
+		return nil, err
+	}
+
+	nativeAst := astNode{}
+	err = json.Unmarshal(sqlAst, &nativeAst)
+	if err != nil {
+		return nil, err
+	}
+
+	ast := &AST{
+		sql:       sql,
+		ast:       nativeAst,
+		rootNodes: make([]*selectNode, 0),
+		aliases:   map[string]bool{},
+		added:     map[string]bool{},
+		fromNodes: make([]*fromNode, 0),
+		columns:   make([]*columnNode, 0),
+	}
+
+	err = ast.traverse()
+	if err != nil {
+		return nil, err
+	}
+	return ast, nil
+}
+
 // Format normalizes a DuckDB SQL statement
-func Format(sql string) (string, error) {
-	return queryString("SELECT json_deserialize_sql(json_serialize_sql(?::VARCHAR))", sql)
+func (a *AST) Format() (string, error) {
+	sql, err := json.Marshal(a.ast)
+	if err != nil {
+		return "", err
+	}
+	res, err := queryString("SELECT json_deserialize_sql(?::JSON)", string(sql))
+	return string(res), err
 }
 
-// Sanitize strips comments and normalizes a DuckDB SQL statement
-func Sanitize(sql string) (string, error) {
-	panic("not implemented")
+func (a *AST) GetTableRef() (*TableRef, bool) {
+	if len(a.fromNodes) == 1 {
+		return a.fromNodes[0].ref, true
+	}
+	return nil, false
+}
```

Comment:
What is the point of this func? (Looks like it will fail if multiple tables are referenced?)

Reply:
This was some prototyping sources as sql. Will remove all related code.